_posts/2024-05-14-registry-dereference.md

```diff
@@ -0,0 +1,285 @@
+props.putIfAbsent(ProducerConfig.CLIENT_ID_CONFIG, "Producer-" + TOPIC_NAME);
+props.putIfAbsent(ProducerConfig.ACKS_CONFIG, "all");
+props.putIfAbsent(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+props.putIfAbsent(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, JsonSchemaKafkaSerializer.class.getName());
+props.putIfAbsent(SerdeConfig.ARTIFACT_RESOLVER_STRATEGY, SimpleTopicIdStrategy.class.getName());
+props.putIfAbsent(SerdeConfig.EXPLICIT_ARTIFACT_GROUP_ID, "default");
+props.putIfAbsent(SerdeConfig.REGISTRY_URL, REGISTRY_URL);
+
+//This is the most important configuration for this particular example.
+props.putIfAbsent(SchemaResolverConfig.SERIALIZER_DEREFERENCE_SCHEMA, "true");
+
+
+// Create the Kafka producer
+Producer<Object, Object> producer = new KafkaProducer<>(props);
+```
+
+With this producer we can then send messages to Kafka that adhere to the _citizen_ schema as follows:
+
+```java
+for(int idx = 0; idx < 5; idx++) {
+    City city=new City("New York",10001);
+    city.setQualification(new CityQualification("city_qualification",11));
+
+    CitizenIdentifier identifier=new CitizenIdentifier(123456789);
+    identifier.setIdentifierQualification(new IdentifierQualification("test_subject",20));
+    Citizen citizen=new Citizen("Carles","Arnal",23,city,identifier,List.of(new Qualification(UUID.randomUUID().toString(),6),new Qualification(UUID.randomUUID().toString(),7),new Qualification(UUID.randomUUID().toString(),8)));
+
+    // Send/produce the message on the Kafka Producer
+    ProducerRecord<Object, Object> producedRecord = new ProducerRecord<>(topicName, SUBJECT_NAME, citizen);
+    producer.send(producedRecord);
+}
+```
+
+And consume them with a consumer that is created as follows, instructing the deserializer to use the schema dereferenced with the following configuration property _DESERIALIZER_DEREFERENCE_SCHEMA_:
+
+```java
+Properties props = new Properties();
+
+props.putIfAbsent(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, SERVERS);
+props.putIfAbsent(ConsumerConfig.GROUP_ID_CONFIG, "Consumer-" + TOPIC_NAME);
+props.putIfAbsent(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true");
+props.putIfAbsent(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, "1000");
+props.putIfAbsent(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+props.putIfAbsent(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+props.putIfAbsent(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, JsonSchemaKafkaDeserializer.class.getName());
+props.putIfAbsent(SerdeConfig.ARTIFACT_RESOLVER_STRATEGY, SimpleTopicIdStrategy.class.getName());
+props.putIfAbsent(SerdeConfig.EXPLICIT_ARTIFACT_GROUP_ID, "default");
+props.putIfAbsent(SerdeConfig.VALIDATION_ENABLED, true);
+
+//This is the most important configuration for this example, that instructs the deserializer to fetch the dereferenced schema.
+props.putIfAbsent(SchemaResolverConfig.DESERIALIZER_DEREFERENCE_SCHEMA, "true");
```

Comment:
String as the value for this property rather than a boolean `true`?

Reply:
It "doesn't matter" in the sense that if the expected value is a boolean it's just parsed, but changing it to better honor the expected value.